ms2deepscore/data_generators.py

```diff
@@ -245,40 +246,37 @@ def __data_generation(self, spectrum_inchikey_ids_batch):
 
         # Generate data
         for i_batch, pair in enumerate(spectrum_inchikey_ids_batch):
-            for i_pair, spectrum_inchikey_id in enumerate(pair):
-                idx, values = self._data_augmentation(self.spectrums_binned[spectrum_inchikey_id[0]].binned_peaks)
+            for i_pair, spectrum_inchikey in enumerate(pair):
+                idx, values = self._data_augmentation(self.spectrums_binned[spectrum_inchikey[0]].binned_peaks)
                 X[i_pair][i_batch, idx] = values
 
-            y[i_batch] = self.score_array[pair[0][1], pair[1][1]]
+            y[i_batch] = self.labels_df[pair[0][1]][pair[1][1]]
 
         return X, y
 
 
 class DataGeneratorAllInchikeys(DataGeneratorAllSpectrums):
     """Generates data for training a siamese Keras model
     This generator will provide training data by picking each training InchiKey
-    listed in *inchikey_ids* num_turns times in every epoch. It will then randomly
+    listed in *selected_inchikeys* num_turns times in every epoch. It will then randomly
     pick one the spectra corresponding to this InchiKey (if multiple) and pair it
     with a randomly chosen other spectrum that corresponds to a reference score
     as defined in same_prob_bins.
     """
-    def __init__(self, spectrums_binned: List[BinnedSpectrum], score_array: np.ndarray,
-                 inchikey_ids: list, inchikey_score_mapping: np.ndarray,
-                 dim: int, **settings):
+    def __init__(self, spectrums_binned: List[BinnedSpectrum], selected_inchikeys: list,
+                 labels_df: pd.DataFrame, dim: int, **settings):
         """Generates data for training a siamese Keras model.
         Parameters
         ----------
         spectrums_binned
             List of BinnedSpectrum objects with the binned peak positions and intensities.
-        score_array
-            2D-array of reference similarity scores (=labels).
-            Must be symmetric (score_array[i,j] == score_array[j,i]) and in the
-            same order as inchikey_score_mapping.
-        inchikey_ids
-            List of IDs from unique_inchikeys to use for training.
-        inchikey_score_mapping
-            Array of all unique inchikeys. Must be in the same order as the scores
-            in scores_array.
+        labels_df
+            Pandas DataFrame with reference similarity scores (=labels) for compounds identified
+            by inchikeys. Columns and index should be inchikeys, the value in a row x column
+            depicting the similarity score for that pair. Must be symmetric
+            (labels_df[i,j] == labels_df[j,i]) and column names should be identical to the index.
```

Comment:
Is this clear enough?

Reply:
Yes, sounds good. (I commented on the term "labels" above, but I'm already getting used to it ;) )